src/docfx/lib/json/YamlUtility.cs

```diff
@@ -181,6 +164,58 @@ private static JToken ParseAsJToken(IParser parser, string file, List<Error> err
             }
         }
 
+        private static JToken ParseScalar(string value)
+        {
+            // https://yaml.org/spec/1.2/2009-07-21/spec.html
+            //
+            //  Regular expression       Resolved to tag
+            //
+            //    null | Null | NULL | ~                          tag:yaml.org,2002:null
+            //    /* Empty */                                     tag:yaml.org,2002:null
+            //    true | True | TRUE | false | False | FALSE      tag:yaml.org,2002:bool
+            //    [-+]?[0 - 9]+                                   tag:yaml.org,2002:int(Base 10)
+            //    0o[0 - 7] +                                     tag:yaml.org,2002:int(Base 8)
+            //    0x[0 - 9a - fA - F] +                           tag:yaml.org,2002:int(Base 16)
+            //    [-+] ? ( \. [0-9]+ | [0-9]+ ( \. [0-9]* )? ) ( [eE][-+]?[0 - 9]+ )?   tag:yaml.org,2002:float (Number)
+            //    [-+]? ( \.inf | \.Inf | \.INF )                 tag:yaml.org,2002:float (Infinity)
+            //    \.nan | \.NaN | \.NAN                           tag:yaml.org,2002:float (Not a number)
+            //    *                                               tag:yaml.org,2002:str(Default)
+            if (string.IsNullOrEmpty(value) || value == "~" || value.Equals("null", StringComparison.OrdinalIgnoreCase))
+            {
+                return JValue.CreateNull();
+            }
+            if (bool.TryParse(value, out var b))
```

Comment:
what about 1/0 for bool? seems you changed the order?

Reply:
1, 0 is not a valid bool, `bool.TryParse("0")` returns `false`